tuna.go

```diff
@@ -311,6 +318,24 @@ func (c *Common) CreateServerConn(force bool) error {
 
 				metadata := c.GetMetadata()
 
+				if !c.EntryServiceInfo.IPFilter.Empty() {
+					if !c.EntryServiceInfo.IPFilter.IPCheck(metadata.IP) {
```

Comment:
We still cannot distinguish whether the ip2c service is down or returned result does not contain location (e.g. unidentified IP). One way to do this is to return (ok, err) from IPCheck. We can also put Empty() inside IPCheck, so the logic becomes:
ok, err = IPCheck()
if err == nil { if !ok then continue  }
else print error but do nothing else